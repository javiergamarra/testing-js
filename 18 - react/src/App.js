import "./App.css";
import { useState } from "react";

function App(props) {

  const [enabled, setEnabled] = useState(false);

  const onClick = () => {
    setEnabled(!enabled);
  };

  const onClickRequest = () => {
    fetch("https://jsonplaceholder.typicode.com/todos/1").then(
      x => {
        console.log(x.json());
      }
    );
  };

  return (<div className="App">

    <button className="button" onClick={onClick} style={{ "fontSize": props.fontSize }}>Enable</button>

    {enabled && (<div>enabled</div>)}

    <input id="name" type="text"></input>

    <p>{props.value}</p>

    <button onClick={onClickRequest}>Call API</button>

  </div>);
}

export default App;
