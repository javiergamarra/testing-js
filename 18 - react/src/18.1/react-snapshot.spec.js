import App from "../App";
import renderer from "react-test-renderer";

describe("testing with snapshots", () => {
  test("appears when clicking the button", async () => {
    const component = renderer.create(<App />);

    expect(component.toJSON()).toMatchSnapshot();

    const button = await component.root.findByType("button");
    button.props.onClick();

    expect(component.toJSON()).toMatchSnapshot();
  });
});