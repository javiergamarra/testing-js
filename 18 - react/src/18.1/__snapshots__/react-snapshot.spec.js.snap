// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing with snapshots appears when clicking the button 1`] = `
<div
  className="App"
>
  <button
    onClick={[Function]}
  >
    Enable
  </button>
</div>
`;

exports[`testing with snapshots appears when clicking the button 2`] = `
<div
  className="App"
>
  <button
    onClick={[Function]}
  >
    Enable
  </button>
  <div>
    enabled
  </div>
</div>
`;
